src/renderer/components/GlobalPage/ConnectionBlock.js

```diff
@@ -52,21 +52,47 @@ export const ConnectionBlock = ({
   const [clusterName, setClusterName] = useState('');
   const [loading, setLoading] = useState(false);
   const [clusterUrl, setClusterUrl] = useState('');
+  const [validationError, setValidationError] = useState(null);
+  const [showInfoBox, setShowInfoBox] = useState(true);
 
-  const checkError = (cloud) => {
+  const checkConnectionError = (cloud) => {
     if (cloud?.connectError) {
       Notifications.error(cloud.connectError);
     }
   };
 
+  const validateCloud = (url) => {
+    if (cloudStore?.clouds?.[url]) {
+      setValidationError(connectionBlock.notice.urlAlreadyUsed());
+      return false;
+    }
+    const validName = clusterName.trim();
+    if (!validName || validName.includes(' ')) {
```

Comment:
Could contain a tab character too, if it was copy/pasted from somewhere else that allowed typing a tab character. I think this would cover all possible cases:

```suggestion
    if (!validName || validName.match(/\s/)) {
```